codefordenver/cdphe-accessible-covid19
# 23 testing visualization

#### What does this PR do?
Addresses [issue 23](https://github.com/codefordenver/cdphe-accessible-covid19/issues/23)

Adds an API client for pulling data from various endpoints. Two requests are made. The first gets data on the number of daily PCR tests administered, while the second gets data on the number of antibody tests. Here is representative output:

PCR:
```
[
    {
        date: "2020-06-08",
        percentPositive: 0.033,
        percentPositive3DayAvg: 0.037,
        percentPositive7DayAvg: 0.046,
        testedAtCommercialLabs: 4445,
        testedAtStateLabs: 388,
        totalTested: 4833,
    }, ...
]
```

Antibody:
```
[
    {
        antibodyTests: 814,
        date: "2020-07-28",
        negativeTests: 753,
        percentPositive: 0.075,
        percentPositive3DayAvg: 0.072,
        percentPositive7DayAvg: 0.063,
        positiveTests: 61,
        testsPer100KPeople: 14.122,
        totalTested: null,
    }, ...
]
```

Also adds a StackedBarPlot component for visualizing daily PCR tests at commercial vs state labs, as well as positive vs. negative daily antibody tests. 

![Screen Shot 2021-01-21 at 7 50 07 PM](https://user-images.githubusercontent.com/20004072/105440624-1ffa2f80-5c24-11eb-82f9-e9e43460ae3d.png)

![Screen Shot 2021-01-21 at 8 06 48 PM](https://user-images.githubusercontent.com/20004072/105440680-3ef8c180-5c24-11eb-86c6-7af08c8f1f15.png)

#### Where should the reviewer start?
The api client functions and the new component are used in App.js. The api client code is in `apiClient.js`, and the visualization is in `StackedBarPlot`. Most of the semiotic code is modified from the documentation [bar plot examples](https://semiotic.nteract.io/guides/bar-chart).

#### How should this be manually tested?
I `console.log`'ed `pcrTestData` and `antibodyTestData` in the `Home` component to make sure the api client works as expected. Then in the browser, I checked the visualizations under the "tested" nav item. You should be able to mouse over the visualizations to see tooltips with more info. 

#### Any background context you want to provide?
Made some changes to other components. I needed the `Card` component to take a `children` prop so that it can display different types of visualizations. If changes need to be made around this, just me know. 

Also tried to pick an accessible color scheme for the bar plots using the ColorBlinding chrome extension, but if that needs to change, just let me know.
 
#### What are the relevant tickets?
Addresses [issue 23](https://github.com/codefordenver/cdphe-accessible-covid19/issues/23)
#### Questions:
